lib/member_page.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'scraped'
+
+# This class represents the profile page of a given member
+class MemberPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::CleanUrls
+
+  field :id do
+    [term, File.basename(url, '.htm')].join('/')
+  end
+
+  field :term do
+    File.dirname(url).split('/').last
+  end
+
+  field :name do
+    name_parts.first.to_s.gsub(Regexp.union(titles << '.'), '').tidy
+  end
+
+  field :honorific_prefix do
+    titles.select { |prefix| name_parts.first.to_s.include? prefix }.join(' ')
+  end
+
+  field :honorific_suffix do
+    name_parts.drop(1).map(&:tidy).join(', ')
+  end
+
+  field :gender do
+    return 'female' if honorific_prefix.include?('Mrs')
+  end
+
+  field :faction do
+    return 'Independent' if (affiliation = political_affiliation).empty?
+    # Some member pages list more than one group affiliation for that member
+    # Here, we remove affiliations with known non-party groups
+    affiliation.map(&:to_s).map(&:tidy).reject do |party|
+      non_party_groups.to_s.include? party
```

Comment:
Why are you casting `non_party_groups` to a string here? The idea of it being a `Set` is that you have an O(1) lookup directly into it…